kong/llm/drivers/cohere.lua

```diff
@@ -114,113 +114,82 @@ local function handle_stream_event(event_string, model_info, route_type)
                   and event.token_count.prompt_tokens
               or 0,
     }
-
   end
 
   if new_event then
-    new_event = cjson.encode(new_event)
+    if new_event ~= "[DONE]" then
+      new_event = cjson.encode(new_event)
+    end
+
     return new_event, nil, metadata
   else
     return nil, nil, metadata  -- caller code will handle "unrecognised" event types
   end
 end
 
-local transformers_to = {
-  ["llm/v1/chat"] = function(request_table, model)
-    request_table.model = model.name
 
-    if request_table.prompt and request_table.messages then
-      return kong.response.exit(400, "cannot run a 'prompt' and a history of 'messages' at the same time - refer to schema")
-  
-    elseif request_table.messages then
-      -- we have to move all BUT THE LAST message into "chat_history" array
-      -- and move the LAST message (from 'user') into "message" string
-      if #request_table.messages > 1 then
-        local chat_history = table_new(#request_table.messages - 1, 0)
-        for i, v in ipairs(request_table.messages) do
-          -- if this is the last message prompt, don't add to history
-          if i < #request_table.messages then
-            local role
-            if v.role == "assistant" or v.role == "CHATBOT" then
-              role = "CHATBOT"
-            else
-              role = "USER"
-            end
-    
-            chat_history[i] = {
-              role = role,
-              message = v.content,
-            }
+local function merge_fields(request_table, model)
```

Comment:
_Continues: https://github.com/Kong/kong/pull/12807/files#r1565697308_

@hanshuebner 

We can't combine these functions, because the output formats are all WILDLY different for each provider.

We are looking into a smarter transformation system, driven by a config file that can be replaced, but that will come later.